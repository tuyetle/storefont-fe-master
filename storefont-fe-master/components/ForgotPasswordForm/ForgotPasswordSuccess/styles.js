import { css } from 'glamor';
import themes from '~/styles/themes';
import variables from '~/styles/variables';

export default {
  sendAgain: css({
    ...themes['font-montserrat-semibold'],
    backgroundColor: 'white',
    border: 'none',
    color: variables.$postListingColor,
    textTransform: 'uppercase',
    fontWeight: 'bold',
    padding: '0.188em',
    textAlign: 'center',
    textDecoration: 'none',
    display: 'inline-block',
    '&:disabled': {
      color: variables.$buttonDisabledColor,
    },
  }),
  count: css({
    color: variables.$buttonDisabledColor,
  }),
  notReceived: css({
    fontSize: '0.938em',
  }),
  tickIcon: css({
    display: 'block',
    margin: '0 auto',
    marginTop: '1.438em',
    marginBottom: '1.375em',
    '@media (min-width: 768px)': {
      marginTop: '1.625em',
      marginBottom: '2em',
    },
  }),
  forgotPasswordSuccess: css({
    ...themes['font-montserrat-semibold'],
    color: '#00a046',
    lineHeight: '1.73',
    fontSize: '0.813em',
    '@media (min-width: 768px)': {
      fontSize: '0.938em',
    },
    textAlign: 'center',
    marginBottom: '2.688em',
  }),
};
