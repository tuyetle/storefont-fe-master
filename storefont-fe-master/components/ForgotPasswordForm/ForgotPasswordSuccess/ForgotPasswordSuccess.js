import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import ReactInterval from 'react-interval';
import { noop } from 'lodash';
import TickIcon from '~/static/assets/img/ic-tick.png';
import { RESUBMIT_TIMEOUT_SIGNUP, TIMEOUT_SIGNUP } from '~/config/config';
import styles from './styles';

class ForgotPasswordSuccess extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { count: RESUBMIT_TIMEOUT_SIGNUP };
  }
  onSubmitAgain = () => {
    this.start();
    this.props.onSubmitAgain();
  }
  submitAgainCallback = () => {
    if (this.state.count !== 0) {
      this.setState({ count: this.state.count - 1 });
    }
  }
  start = () => {
    this.setState({ count: RESUBMIT_TIMEOUT_SIGNUP });
  }
  render() {
    const allowSendAgain = this.state.count !== 0;
    return (
      <Fragment>
        <img className={styles.tickIcon} src={TickIcon} alt={this.props.t('sendNewPassword')} />
        <div className={styles.forgotPasswordSuccess}>
          {this.props.t('resetPasswordNotificationBefore')}<br />
          {this.props.t('resetPasswordNotificationAfter')}
        </div>
        <div className="text-center">
          <span className={styles.notReceived}>{this.props.t('notReceived')}</span>
          <button
            className={styles.sendAgain}
            onClick={this.onSubmitAgain}
            disabled={allowSendAgain}
          >
            {this.props.t('sendAgain')}
          </button>
          {
            allowSendAgain &&
              <span className={styles.count}>({this.state.count})</span>
          }
          <ReactInterval
            timeout={TIMEOUT_SIGNUP}
            callback={this.submitAgainCallback}
            enabled={allowSendAgain}
          />
        </div>
      </Fragment>
    );
  }
}

ForgotPasswordSuccess.propTypes = {
  onSubmitAgain: PropTypes.func,
  t: PropTypes.func,
};
ForgotPasswordSuccess.defaultProps = {
  onSubmitAgain: noop,
  t: noop,
};
export default translate(['auth'])(ForgotPasswordSuccess);

