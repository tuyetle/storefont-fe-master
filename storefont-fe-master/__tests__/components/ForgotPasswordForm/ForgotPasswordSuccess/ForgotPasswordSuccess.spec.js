import React from 'react';
import ForgotPasswordSuccess from '~/components/ForgotPasswordForm/ForgotPasswordSuccess/ForgotPasswordSuccess';
import renderer from 'react-test-renderer';

describe('(Component) ForgotPasswordSuccess', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      t(value) { return value; },
    };
    component = mount(<ForgotPasswordSuccess {...props} />);
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
