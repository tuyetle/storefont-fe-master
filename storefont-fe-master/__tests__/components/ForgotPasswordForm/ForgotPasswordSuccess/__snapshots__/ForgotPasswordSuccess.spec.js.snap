// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ForgotPasswordSuccess should render conrectly 1`] = `
Array [
  <img
    alt="sendNewPassword"
    className={
      Object {
        "data-css-1i7cb0r": "",
      }
    }
    src="/static/static/assets/img/ic-tick.png"
  />,
  <div
    className={
      Object {
        "data-css-19amwto": "",
      }
    }
  >
    resetPasswordNotificationBefore
    <br />
    resetPasswordNotificationAfter
  </div>,
  <div
    className="text-center"
  >
    <span
      className={
        Object {
          "data-css-fzywh6": "",
        }
      }
    >
      notReceived
    </span>
    <button
      className={
        Object {
          "data-css-1qmlgfq": "",
        }
      }
      disabled={true}
      onClick={[Function]}
    >
      sendAgain
    </button>
    <span
      className={
        Object {
          "data-css-14wv70v": "",
        }
      }
    >
      (
      59
      )
    </span>
  </div>,
]
`;
